File: ALG_lista2/ex2.py
def adicionarCarro(estacionamento):
    andar = int(input("Andar: "))
    vaga = int(input("Vaga: "))

    if estacionamento[andar][vaga] == 1:
        print("Esta vaga ja esta preenchida\n");
        return
    else:
        estacionamento[andar][vaga] = 1

def removerCarro(estacionamento):
    andar = int(input("Andar: "))
    vaga = int(input("Vaga: "))

    if estacionamento[andar][vaga] == 0:
        print("Esta vaga ja esta vazia\n");
        return
    else:
        estacionamento[andar][vaga] = 0

def vagasPorAndar(estacionamento):
    qtd = 0

    andar = int(input("Qual o andar: "))

    for i in range(10):
        if estacionamento[andar][i] == 0:
            print(f"[{i}] vaga livre")
            qtd += 1
    print(f"Total = {qtd}")

def vagasEstacionamento(estacionamento):
    total = 0

    print("Andar '0'");
    for i in range(10):
        if estacionamento[0][i] == 0:
            print(f"[{i}] livre")
            total += 1

    print("Andar '1'");
    for i in range(10):
        if estacionamento[1][i] == 0:
            print(f"[{i}] livre")
            total += 1
    
    print(f"\nVagas livres totais = {total}");

estacionamento = [[0]*10 for i in range(2)]

while True:
    print("1 - adicionar carro");
    print("2 - remover carro");
    print("3 - vagas por andar");
    print("4 - vagas estacionamento");
    print("0 - sair\n");

    opcao = int(input("Escolha uma opcao: "))

    match opcao:
        case 1:
            adicionarCarro(estacionamento)
        case 2:
            removerCarro(estacionamento)
        case 3:
            vagasPorAndar(estacionamento)
        case 4:
            vagasEstacionamento(estacionamento)
        case 0:
            print("Encerrando sistema")
            break
        case _:
            print("Opcao invalida!")
            break

File: ALG_lista2/ex2.c
#include <stdio.h>

void adicionarCarro(int estacionamento[2][10]){
    int andar, vaga;

    printf("\nQual andar e vaga deseja adicionar seu carro: \n");
    scanf("%d %d", &andar, &vaga);

    if(estacionamento[andar][vaga] == 1){
        printf("Esta vaga ja esta preenchida\n");
        return;
    } else {
        estacionamento[andar][vaga] = 1;
    }
}

void removerCarro(int estacionamento[2][10]){
    int andar, vaga;

    printf("\nQual andar e vaga deseja remover seu carro: \n");
    scanf("%d %d", &andar, &vaga);

    if(estacionamento[andar][vaga] == 0){
        printf("Nao tem nenhum carro nessa vaga\n");
    } else {
        estacionamento[andar][vaga] = 0;
    }
}

void vagasPorAndar(int estacionamento[2][10]){
    int andar;
    int qtd = 0;

    printf("Qual andar: ");
    scanf("%d", &andar);

    for(int i = 0; i < 10; i++){
        if(estacionamento[andar][i] == 0){
            printf("[%d] livre\n", i);
            qtd++;
        }
    }
    printf("\nQuantidade = %d\n", qtd);
}

void vagasEstacionamento(int estacionamento[2][10]){
    int total = 0;

    printf("Andar '0'\n");
    for(int i = 0; i < 10; i++){
        if(estacionamento[0][i] == 0){
            printf("[%d] livre\n", i);
            total++;
        }
    }

    printf("\nAndar '1'\n");
    for(int i = 0; i < 10; i++){
        if(estacionamento[1][i] == 0){
            printf("[%d] livre\n", i);
            total++;
        }
    }
    printf("\nVagas totais = %d\n", total);
}

int main(){

    int estacionamento[2][10] = {0};
    int opcao;

    do {
        printf("1 - adicionar carro\n");
        printf("2 - remover carro\n");
        printf("3 - vagas por andar\n");
        printf("4 - vagas estacionamento\n");
        printf("0 - sair\n");

        printf("\nEscolha uma opcao: ");
        scanf("%d", &opcao);

        switch (opcao){
            case 1:
                adicionarCarro(estacionamento);
                break;

            case 2:
                removerCarro(estacionamento);
                break;

            case 3:
                vagasPorAndar(estacionamento);
                break;

            case 4:
                vagasEstacionamento(estacionamento);
                break;

            case 0:
                printf("\nEncerrando sistema\n");
                break;

            default:
                printf("Opcao invalida\n");
                return 1;
        }

    } while (opcao != 0);

    return 0;
}